source/unsafe-messenger.js

```diff
@@ -73,6 +74,34 @@ export default function unsafeMessenger() {
 			editor.blur();
 		});
 	}
+
+	function monacoEditor(target) {
+		const data_uri = target.getAttribute("data-uri");
+		const editor = monaco.editor.getModels(data_uri)[0];
```

Comment:
Maybe it's this?

```suggestion
		const editor = monaco.editor.getModels(data_uri).find(model => model._domElement === target)
```

Check the `getModels()` array there's anything to match the model to the `target`. I think it's the `_domElement` property

Reply:
You can see this issue on Microsoft’s own demo page

![ko](https://user-images.githubusercontent.com/1402241/215008693-48d00ce7-af1c-4bab-92da-13b4e7632e5f.gif)

Reply:
Oh ignore my previous comment. 

`getModels` does not accept a parameter, it always gets all the models:

https://microsoft.github.io/monaco-editor/api/modules/monaco.editor.html#getModels

You're looking for:

```suggestion
		const editor = monaco.editor.getModel(data_uri);
```